### PR TITLE
No more PG dependency

### PR DESCRIPTION
**Pull request by expede**

### CODE REVIEW

**Comment by dholms** [comment]
Still segfaults :sob: 
buuutttt, now there's no dependencies that need to be installed, so we can just distribute the dynamic binary?
Use Azure pipelines to build for older versions of Linux?

**Comment by expede** [comment]
> buuutttt, now there's no dependencies

It still depends on `libtinfo`, which is a very widely used dynamic library (e.g. it's a dependency for `ncurses`). A lot of machines will have it already, and adding it to an Aptitude build script would probably be sufficient.

The static binary thing is going to drive me wild, so I'm trying some things under virtualization. Some Googling suggests that enabling the `fPIC` GCC flag can help. It tells GCC to not depend on code being in any particular location (`--ghc-options '-optl-static fPIC'). Anyhow, I'll plug away at this for a bit and see

**Comment by expede** [comment]
🤔

```
ghc-options: -Wall -O2 -static -threaded
cc-options: -static
ld-options: -static -pthread
extra-lib-dirs: ./.system-work/lib
```

https://www.reddit.com/r/haskell/comments/9on8gi/is_there_an_easy_way_to_compile_static_binaries/

**Comment by expede** [comment]
To more directly address older Linuxes

> Use Azure pipelines to build for older versions of Linux?

I think we should see if you can run something compiled on the last LTS (18.04). Otherwise, we can set a standard of the last two LTSes, plus the latest version (19.10), plus _maybe_ something for Fedora. As it gets increasingly obscure, people can either install from source or write their own release packages. At least I've seen that as a strategy a lot.

And you're right that having minimal dependencies (in this case libtinfo) is totally viable. I have a BUNCH of libtinfo-dependant packages installed through Brew.

**Comment by expede** [comment]
You know, I thought it through, and did some research. It's actually fairly rare to have a fully static binary, and platform package managers (apt, yum, brew) tend to install dynamic dependencies for you. As much as a fully static binary would be nice, it's a huge PITA. The easiest thing is to add more to the Azure build pipeline, and write a few installer scripts, and update them every few months/years.

Still looking for PR approval on this one, to build dynamic binaries without the PG dependency

File: library/Fission/Web/Client/Register.hs [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
+  -> ClientM ()
+
+verify
+  :: BasicAuthData
+  -> ClientM Bool
 
-register :<|> verify = client (Proxy :: Proxy UserRoute)
+reset
+  :: BasicAuthData
+  -> Fission.Web.User.Password.Reset.Types.Reset
```
Comment (bechurch) [inline library/Fission/Web/Client/Register.hs]: Given the above import
```haskell
import           Fission.Web.User.Password.Reset.Types
```
shouldn't this line just be
```haskell
-> Reset
```
?

**Comment by expede** [comment]
@bechurch the change is upstream: https://github.com/fission-suite/web-api/pull/203

File: library/Fission/Web/Client/Register.hs (hunk 1 above)
Comment (expede) [reply library/Fission/Web/Client/Register.hs]: Ah, indeed. I hadn't pushed that 😅 Pushing now

Comment (expede) [reply library/Fission/Web/Client/Register.hs]: Same goes for `Password`

**Comment by expede** [comment]
Sure thing!

Ugh, I wish GH had threaded comments in PRs

**Pull request merged**